Optimize reading blobs on SQL Server

OronDF343 opened the pull request:
Fixes #2054

Use `CommandBehavior.SequentialAccess` to work around performance issues with SqlClient when reading blobs (SQL Server specific), and remove unnecessary copying to `MemoryStream` (All ADO.NET delegates).

OronDF343 commented:
Test failure appears to be unrelated to changes, not sure why it is happening

lahma commented:
Something with this one seems to have a problem, can you use the same logic as in 3.x without the job builder?

OronDF343 commented:
Doesn't seem to have helped... The faked DbDataReader is giving empty columns

lahma commented:
Thanks for checking, I'll try to have a look later.

OronDF343 commented:
I found what is causing the test to fail. In net472 and in version 3.x, the extension method `Quartz.Util.DataReaderExtensions.GetString` is used, while in `net6.0`, `System.Data.DataReaderExtensions.GetString` is used. The custom extension uses the indexer `[]` while the system extension apparently does not, which means that the faked call to the indexer is not effective.

The reason why the matched method has changed is due to adding a `net6.0` target to Quartz, since that method does not exist in `netstandard2.0` or `net472`. Also, since it is an extension method, it cannot be directly faked.

I have reverted to using the builder and fixed the failing test for now.

lahma commented on src/Quartz/Impl/AdoJobStore/StdAdoDelegate.Calendars.cs:
```diff
@@ -1,4 +1,5 @@
 using Microsoft.Extensions.Logging;
+using System.Data;
```
the integration tests seem to be showing a real problem, now different extension method is being chosen for `reader.GetString`, I think you need to remove `using System.Data;` additions in this PR and instead refer to `SequentialAccess` with its full name `System.Data.CommandBehavior.SequentialAccess`

lahma replied:
I couldn't easily push a tweak to to this PR as you have created your PR for main - it's better convention to have PR's made from separate branch

OronDF343 replied:
Ok, that makes a lot more sense. I am removing the usings that I added now and rolling back the test

lahma commented:
Nice detective work, the approach still has problem (and now easy to understand why as you checked the extension method). I've commented about the problem and why integration tests are now failing.

OronDF343 commented:
Alright, looks like it's passing now, thanks for the feedback

lahma commented:
Thanks for getting this ready for `main` too!
